contracts/shapella-upgrade/ShapellaUpgradeTemplate.sol

```diff
@@ -0,0 +1,374 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.9;
+
+
+interface IAccessControl {
+    function DEFAULT_ADMIN_ROLE() external returns (bytes32);
+    function grantRole(bytes32 role, address account) external;
+    function renounceRole(bytes32 role, address account) external;
+}
+
+interface IAccountingOracle {
+    function SUBMIT_DATA_ROLE() external view returns (bytes32);
+    function initialize(address admin, address consensusContract, uint256 consensusVersion) external;
+}
+
+interface IBaseOracle {
+    function getConsensusContract() external view returns (address);
+}
+
+interface IBurner is IAccessControl {
+    function REQUEST_BURN_SHARES_ROLE() external view returns (bytes32);
+}
+
+interface IHashConsensus {
+    function MANAGE_MEMBERS_AND_QUORUM_ROLE() external view returns (bytes32);
+    function addMember(address addr, uint256 quorum) external;
+    function getFrameConfig() external view returns (uint256 initialEpoch, uint256 epochsPerFrame, uint256 fastLaneLengthSlots);
+    function updateInitialEpoch(uint256 initialEpoch) external;
+}
+
+interface ILido {
+    function finalizeUpgrade_v2(address _lidoLocator, address _eip712StETH) external;
+}
+
+interface ILidoLocator {
+    function accountingOracle() external view returns(address);
+    function depositSecurityModule() external view returns(address);
+    function elRewardsVault() external view returns(address);
+    function legacyOracle() external view returns(address);
+    function lido() external view returns(address);
+    function oracleReportSanityChecker() external view returns(address);
+    function burner() external view returns(address);
+    function stakingRouter() external view returns(address);
+    function treasury() external view returns(address);
+    function validatorsExitBusOracle() external view returns(address);
+    function withdrawalQueue() external view returns(address);
+    function withdrawalVault() external view returns(address);
+    function postTokenRebaseReceiver() external view returns(address);
+    function oracleDaemonConfig() external view returns(address);
+}
+
+interface ILegacyOracle {
+    function finalizeUpgrade_v4(address _accountingOracle) external;
+    function getContractVersion() external view returns (uint256);
+}
+
+interface ILidoOracle {
```

Comment:
Nope, genuine `ILidoOracle`

Reply:
then `ILegacyOracle` maybe

Reply:
nope, because ILegacyOracle is the new contract, which doesn't have the functions. The "new" and "old" semantics kind of swapped here, so it's not very clear, but at least it's correct and corresponds to the actual contracts.